### PR TITLE
SymbolTable: provide more detailed information about re-defined symbols.

### PR DESCRIPTION
**Pull request by hzeller**
Provide location of new and previous definition of a symbol that
is found out to be re-defined.

Signed-off-by: Henner Zeller <hzeller@google.com>

### CODE REVIEW

File: common/text/text_structure.cc [hunk 1]
```diff
@@ -132,10 +132,23 @@ LineColumnRange TextStructureView::GetRangeForToken(
...
   return {GetLineColAtOffset(token.left(Contents())),
           GetLineColAtOffset(token.right(Contents()))};
 }
 
+LineColumnRange TextStructureView::GetRangeForText(
+    absl::string_view text) const {
+  const int from = std::distance(Contents().begin(), text.begin());
+  const int to = std::distance(Contents().begin(), text.end());
+  CHECK_GE(from, 0) << '"' << text << '"';
+  CHECK_LT(to, Contents().length()) << '"' << text << '"';
```
Comment (mglb) [inline common/text/text_structure.cc:148]: `CHECK_LE`? There's a possibility that `text.end() == Contents().end()`.

File: common/text/text_structure_test.cc [hunk 2]
```diff
@@ -216,6 +216,25 @@ TEST_F(TokenRangeTest, GetRangeOfTokenVerifyAllRangesExclusive) {
...
+  const TokenInfo& token = data_.FindTokenAt({0, 7});
+  EXPECT_EQ(token.text(), "world");
+  const LineColumnRange token_range = data_.GetRangeForToken(token);
+  EXPECT_EQ(token_range.start.line, 0);
+  EXPECT_EQ(token_range.start.column, 7);
+
+  const LineColumnRange text_range = data_.GetRangeForText(token.text());
+  EXPECT_EQ(text_range.start.line, 0);
+  EXPECT_EQ(text_range.start.column, 7);
+}
```
Comment (mglb) [inline common/text/text_structure_test.cc:236]: Let's add test case that fails due to CHECK_LE problem mentioned in the other comment:
```
data_.GetRangeForText(data_.Contents());
```

**Review by mglb — CHANGES_REQUESTED** [review]

File: common/text/text_structure.cc (hunk 1 above)
Comment (hzeller) [reply common/text/text_structure.cc]: Oops, that was a leftover from some experiment. Fixed and changed to `CHECK_LE()`

File: common/text/text_structure_test.cc (hunk 2 above)
Comment (hzeller) [reply common/text/text_structure_test.cc:247]: Done.

**Review by mglb — APPROVED** [review]

**Pull request merged**